pkg/rebalancer/storage/controller/service/discover.go

```diff
@@ -0,0 +1,322 @@
+							MemoryLimit: p.MemLimit,
+							MemoryUsage: mpod.Mem,
+						})
+					}
+				}
+
+				jobs, ok = d.jobs.Load().([]job.Job)
+				if !ok {
+					log.Info("job is empty")
+					continue
+				}
+				jobModels = make(map[string][]*model.Job)
+				for _, j := range jobs {
+					var t time.Time
+					if j.Status.StartTime != nil {
+						t = j.Status.StartTime.Time
+					}
+					if _, ok := jobModels[j.Namespace]; !ok {
+						jobModels[j.Namespace] = make([]*model.Job, 0)
+					}
+
+					jobModels[j.Namespace] = append(jobModels[j.Namespace], &model.Job{
+						Name:                 j.Name,
+						Namespace:            j.Namespace,
+						Active:               j.Status.Active,
+						StartTime:            t,
+						Type:                 j.Labels["type"],
+						TargetAgentNamespace: j.Labels["target_agent_namespace"],
+						TargetAgentName:      j.Labels["target_agent_name"],
+						ControllerNamespace:  j.Labels["controller_namespace"],
+						ControllerName:       j.Labels["controller_name"],
+					})
+				}
+
+				// TODO: cache specified reconciled result based on agentResourceType.
+				switch d.agentResourceType {
+				case "statefulset":
+					ss, ok = d.statefulSets.Load().(statefulset.StatefulSet)
+					if !ok {
+						log.Info("statefulset is empty")
+						continue
+					}
+
+					ssModel[ss.Namespace] = &model.StatefulSet{
+						Name:            ss.Name,
+						Namespace:       ss.Namespace,
+						DesiredReplicas: ss.Spec.Replicas,
+						Replicas:        ss.Status.Replicas,
+					}
+				default:
+					// TODO: define error for return
+					return nil
+				}
+
+				// TODO: export below logic to other internal function.
+				var (
+					mmu        map[string]float64 = make(map[string]float64) // max memory usage
+					amu        map[string]float64 = make(map[string]float64) // average memory usage
+					rate       map[string]float64 = make(map[string]float64) // rabalance rate
+					maxPodName map[string]string  = make(map[string]string)
+				)
+				if prevSsModel != nil {
+					for ns, psm := range prevSsModel {
+						if _, ok := ssModel[ns]; !ok {
+							continue
+						}
+						if psm.Replicas > ssModel[ns].Replicas {
+							// TODO: Check the difference prevPodModels and podModels
+							// TODO: create job
+						} else {
+							for _, p := range podModels[ns] {
+								u := p.MemoryUsage / p.MemoryLimit
+								amu[ns] += u
+								if u > mmu[ns] {
+									mmu[ns] = u
+									maxPodName[ns] = p.Name
+								}
+							}
+							amu[ns] = amu[ns] / float64(len(podModels[ns]))
+						}
+						bias := mmu[ns] - amu[ns]
+						if bias > d.tolerance {
+							rate[ns] = 1 - (amu[ns] / mmu[ns])
+							for _, jobs := range jobModels {
+								var ok bool
+								for _, job := range jobs {
+									if job.Type == "rebalance" && job.Active != 0 && job.TargetAgentNamespace == ns {
+										ok = true
+										break
+									}
+								}
+
+								if !ok {
+									// TODO: create Job
+								}
+							}
```

Comment:
This logic will create same job because each this loop ignores the jobs which are created by another difference namespace controller.